gprofiler/utils.py

```diff
@@ -338,19 +344,36 @@ def log_system_info():
     logger.info(f"Total CPUs: {os.cpu_count()}")
     logger.info(f"Total RAM: {psutil.virtual_memory().total / (1 << 30):.2f} GB")
 
-    results = []
+    distribution = "unknown"
+    libc_version = "unknown"
 
     # move to host mount NS for distro & ldd.
     # now, distro will read the files on host.
-    def get_distro_and_libc():
-        results.append(distro.linux_distribution())
-        results.append(get_libc_version())
+    # also move to host UTS NS for the hostname.
+    def get_infos():
+        nonlocal distribution, libc_version
+        global hostname
+
+        try:
+            distribution = distro.linux_distribution()
+        except Exception:
+            logger.exception("Failed to get distribution")
+
+        try:
+            libc_version = get_libc_version()
+        except Exception:
+            logger.exception("Failed to get libc version")
 
-    run_in_ns(["mnt"], get_distro_and_libc)
-    assert len(results) == 2, f"only {len(results)} results, expected 2"
```

Comment:
We previously raised an error when we didn't get all of the expected values, but now you are only logging them and using `unknown` as a replacement value. Is this the new desired behavior?

Reply:
Yep. This function is known to fail in certain cases (for example, there's no `ldd` executable/script in Google Co-OS so that part always fails). The different data items really are unrelated so I prefer that each can fail separately, raise its exception separately and use a placeholder for "error".

Reply:
OK, so do you think it's fine to display a hostname filter with `unknown`? It might be better to handle that specific case

Reply:
Oh, in the backend we should do it, yes. Was planning to after this PR.

What we can instead do is use a more specific "error" hostname, like `<unknown>` (which is less likely to happen in real life)

Reply:
Changed to `<unknown>`